levels/levels.go

```diff
@@ -68,7 +68,17 @@ func (c *Challenge) AddLevel(level Level) {
 
 func (c *Challenge) CheckCurrentLevel() bool {
 	level := c.IDToIndex(*c.CurrentLevel)
-	return CmdOK(c.Levels[level].TestCmd)
+	passed, next_level := CmdOK(c.Levels[level].TestCmd)
+	if next_level == "" {
+		return passed
+	} else {
+		index := c.LevelNameToIndex(next_level)
+		if index == -1 {
+			return false
+		}
+		*c.CurrentLevel = IndexToID(index, c.Name)
```

Comment:
@pepol That is the intended behavior. You set level to `X`. That is the same as if you would have passed the test for the level `X`. The text for `X` will be printed and you will end up in level `X+1`.

It would be difficult to do it any other way, given the simple architecture of this level processing.